config/Dockerfiles/fedoraci-runner/rpmbuild/koji_build_pr.sh

```diff
@@ -58,34 +58,48 @@ kinit -k -t "${CURRENTDIR}/fedora.keytab" $FEDORA_PRINCIPAL
 export FORCE_UNSAFE_CONFIGURE=1
 
 # Build the package with koji
-koji ${KOJI_PARAMS} build --wait --arch-override=x86_64 --scratch ${branch} ${fed_repo}*.src.rpm | tee ${LOGDIR}/kojioutput.txt
-# Set status if either job fails to build the rpm
-RPMBUILD_RC=$?
-if [ "$RPMBUILD_RC" != 0 ]; then
-     echo "status=FAIL" >> ${LOGDIR}/job.props
-     echo -e "ERROR: KOJI BUILD\nSTATUS: $RPMBUILD_RC"
-     exit 1
-fi
+# ignore koij exit status. If build fails it will be detected later, https://pagure.io/fedora-ci/general/issue/76
+koji ${KOJI_PARAMS} build --wait --arch-override=x86_64 --scratch ${branch} ${fed_repo}*.src.rpm | tee ${LOGDIR}/kojioutput.txt || true
 
 popd
 
 SCRATCHID=$(cat ${LOGDIR}/kojioutput.txt | awk '/Created task:/ { print $3 }')
+if ! [[ $SCRATCHID =~ ^[0-9]+$ ]]; then
+    echo "status=FAIL" >> ${LOGDIR}/job.props
+    echo -e "ERROR: KOJI BUILD"
+    exit 1
+fi
+
 echo "koji_task_id=${SCRATCHID}" >> ${LOGDIR}/job.props
 
 # Make sure koji build finished
 # https://pagure.io/fedora-ci/general/issue/76
-koji taskinfo ${SCRATCHID} | tee ${LOGDIR}/taskinfo.txt
+for i in {1..5}; do
+    if koji taskinfo ${SCRATCHID} | tee ${LOGDIR}/taskinfo.txt; then
+        break
+    fi
+    if [[ $i -eq 5 ]]; then
+        echo "status=FAIL" >> ${LOGDIR}/job.props
+        echo -e "ERROR: KOJI TASK_INFO"
+        exit 1
+    fi
+    sleep 60
+done
 TASK_STATE=$(cat ${LOGDIR}/taskinfo.txt | grep "State:" | awk '{print$2}')
```

Comment:
What if the entire block of code is replaced with:

    TASK_STATE=unknown

Reply:
it is possible, but I'd like to avoid getting stuck in the while forever if for some reason the task id is invalid.

Reply:
not sure if it would be ever possible, but I just don't like the possibility ;)

Reply:
If I read the code correctly, the while loop will exit in such case after 5 minutes.

Reply:
yes, sorry about after reading the code again it looks safe to do it